### PR TITLE
add `listclients` functionality

### PR DESCRIPTION
**Pull request by kouloumos**
This PR allows the user to list the nostr clients that are connected to a civikit-node using the `civkit-cli listclients` command.

### Overview

The list of connected clients is part of `ClientHandler`. Instead of making that a shared state and pass it to `ServiceManager`, I used another approach (actors) which [seems](https://ryhl.io/blog/actors-with-tokio/) to [be popular](https://eo-rs.dev/blog/the-actor-pattern-with-async-rust/) for this kind of flows; from what I understand, `ClientHandler` already matches the description of an actor:
> "**The basic idea behind an actor** is to spawn a self-contained task that performs some job independently of other parts of the program. Typically these actors communicate with the rest of the program through the use of message passing channels. [...] A common use-case of actors is to assign the actor exclusive ownership of some resource you want to share, and then let other tasks access this resource indirectly by talking to the actor. " ~ [Actors with Tokio](https://ryhl.io/blog/actors-with-tokio/)

 This approach involves utilizing a channel between the requester and the responder(actor) to send a message that includes a one-shot channel through which the actor responds. We already have a `ServiceManager`->`ClientHandler` channel that sends `ClientEvents`, so I added a `GetClients` message with which the `ServiceManager` requests the clients' information alongside the one-shot channel
through which the `ClientHandler` responds.

### Test the change

1. Start a `civkitd` instance alongside two instance of `civkit-sample`.
2. From one of the `civkit-sample` instances, `opensubscription` and `sendtextnote`.
3. Preview of  `./civkit-cli listclients` command:
  ```
 [CIVKIT-CLI] clients [
    Client {
        pubkey: "ae065e37d162f14cfecd367a8717deebb47996024a5c81a5332861d66924ab18",
        client_id: 1,
        associated_socket: "[::1]:56912",
        subscriptions: 1,
    },
    Client {
        pubkey: "",
        client_id: 2,
        associated_socket: "[::1]:56918",
        subscriptions: 0,
    },
]
  ```

### CODE REVIEW

File: src/proto/boardctrl.proto [hunk 1]
```diff
@@ -43,8 +43,15 @@ message ReceivedNote {
 message ListClientRequest {
 }
 
+message Client {
```
Comment (ariard) [inline src/proto/boardctrl.proto:46]: feel free to add comments for new additions in `boardctrl.proto`, if we start to split into multi process the main components in the future, the interface will be reused quite

**Review by ariard — APPROVED** [review]
Tested ACK e0d53801

> This approach involves utilizing a channel between the requester and the responder(actor) to send a message that includes a one-shot channel through which the actor responds. 

100% the actor model was quite followed as a design pattern, even if the separation between the tasks and the channels is still early at that stage.

**Comment by ariard** [comment]
I’ll take it, it has been tested locally and works okay on my side.

**Comment by kouloumos** [comment]
Please do not hold back any comments that you might have. Always with respect to your review time, I would appreciate any feedback on my code, as this is my first time writing Rust.

**Pull request merged**